### PR TITLE
Update javax.mail to 1.6.5

### PR DESCRIPTION
**Pull request by jvz**
This has changed maven coordinates to the jakarta project, though the
code is binary compatible still.

### Proposed changelog entries

* _Developer:_ Update javax.mail to jakarta.mail 1.6.5

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/pom.xml
Comment (timja) [inline core/pom.xml:336]: why is this removed?

Comment (jvz) [reply core/pom.xml:336]: Good question. I'm not even sure what was patched originally, but it's a transitive dependency of jakarta.mail already.

**Review by timja — APPROVED** [review]
do we need to run PCT or full ATH on this?

edit: I guess if binary compat tests pass, it should be fine

**Comment by timja** [comment]
We probably need a change log entry for this? in case it causes issues

**Comment by jvz** [comment]
> do we need to run PCT or full ATH on this?

I ran PCT of this using a branch from `stable-2.222` with no issues. It looks like ATH is automatically run for PRs in this repo.

As for the javax.activation thing, this looks like the fork, but it has no history: https://github.com/kohsuke/javax.activation

**Review by jeffret-b — APPROVED** [review]
Seems good, if it's sufficiently tested. I haven't testing it any myself.

**Comment by timja** [comment]
Only ath smoke tests are run on PR not the full set afaik

**Comment by res0nance** [comment]
Full changelog [here](https://eclipse-ee4j.github.io/mail/docs/CHANGES.txt)

**Review by res0nance — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Review by oleg-nenashev — COMMENTED** [review]
Be careful with Jakarta code for the `activation` lib, there are/were known issues with module handling there due to the groupId changes. Before we merge that, an investigation is needed. Maybe a manual testing on Java 11 also

**Comment by jvz** [comment]
Thanks, I was wondering why that was patched! If this release is wonky, we'll need to create another update for a forked version of that.

**Comment by jvz** [comment]
I tested this out locally on Java 11 using Gmail as my SMTP host (had to disable a security setting that marks Jenkins as an "insecure" client to try it out), worked fine. Only log messages I'm seeing are related to a removal of javax.annotation stuff.

**Comment by jvz** [comment]
Test failures are unrelated it looks like. Infrastructural.

**Comment by jeffret-b** [comment]
This will be ready for release after the customary waiting period of 24 hours. However, we may want to wait until after the release of 2.235 based upon the current [LTS selection discussion](https://groups.google.com/forum/#!topic/jenkinsci-dev/yiFsJSAKGso).

**Comment by timja** [comment]
> This will be ready for release after the customary waiting period of 24 hours. However, we may want to wait until after the release of 2.235 based upon the current [LTS selection discussion](https://groups.google.com/forum/#!topic/jenkinsci-dev/yiFsJSAKGso).

I agree, unless any objection jvz can we skip this from the current weekly as I assume there's no rush on this?

**Comment by jvz** [comment]
There are some security hardening updates included in newer javamail releases which would be great to get into an LTS.

**Comment by timja** [comment]
Sure, easy enough to back out if any issues then

**Comment by jeffret-b** [comment]
Matt raises a good point so I retract my earlier comment about waiting.

**Pull request merged**
